src/ScriptCs.Core/ScriptLibraryComposer.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Common.Logging;
+using ScriptCs.Contracts;
+using System.IO;
+
+namespace ScriptCs
+{
+    public class ScriptLibraryComposer : IScriptLibraryComposer
+    {
+        private readonly IFileSystem _fileSystem;
+        private readonly IFilePreProcessor _preProcessor;
+        private readonly IPackageContainer _packageContainer;
+        private readonly IPackageAssemblyResolver _packageAssemblyResolver;
+        private readonly ILog _logger;
+
+        public ScriptLibraryComposer(IFileSystem fileSystem, IFilePreProcessor preProcessor, IPackageContainer packageContainer, IPackageAssemblyResolver packageAssemblyResolver, ILog logger)
+        {
+            Guard.AgainstNullArgument("fileSystem", fileSystem);
+            Guard.AgainstNullArgument("preProcessor", preProcessor);
+            Guard.AgainstNullArgument("packageContainer", packageContainer);
+            Guard.AgainstNullArgument("packageAssemblyResolver", packageAssemblyResolver);
+            Guard.AgainstNullArgument("logger", logger);
+
+            _fileSystem = fileSystem;
+            _preProcessor = preProcessor;
+            _packageContainer = packageContainer;
+            _packageAssemblyResolver = packageAssemblyResolver;
+            _logger = logger;
+        }
+
+        internal string GetMainScript(IPackageObject package)
+        {
+            var script =
+                package.GetContentFiles()
+                    .SingleOrDefault(f => f.EndsWith("main.csx", StringComparison.InvariantCultureIgnoreCase));
+            
+            if (script != null)
+            {
+                _logger.DebugFormat("Found main script: {0}", script);    
+            }
+            
+            return script;
+        }
+
+        public virtual string ScriptLibrariesFile
+        {
+            get { return "ScriptLibraries.csx"; }
+        }
+
+        public void Compose(StringBuilder builder = null)
+        {
+            var namespaces = new List<string>();
+            var references = new List<string>();
+
+            var packagesPath = Path.Combine(_fileSystem.CurrentDirectory, _fileSystem.PackagesFolder);
+            var packageReferences = _packageAssemblyResolver.GetPackages(_fileSystem.CurrentDirectory);
+            var packageScriptsPath = Path.Combine(packagesPath, ScriptLibrariesFile);
+
+            if (!_fileSystem.DirectoryExists(packagesPath) || _fileSystem.FileExists(packageScriptsPath))
```

Comment:
what happens if I:
1) install script lib from nuget, run scriptcs
2) install another script lib afterwards
It seems like it will not be picked up because a packageScriptsPath will already exist?